crates/viewer/re_view_spatial/src/visualizers/cylinders3d.rs

```diff
@@ -0,0 +1,286 @@
+use std::iter;
+
+use re_types::{
+    Archetype as _, ArrowString,
+    archetypes::Cylinders3D,
+    components::{ClassId, Color, FillMode, HalfSize3D, Length, Radius, ShowLabels},
+};
+use re_view::clamped_or_nothing;
+use re_viewer_context::{
+    IdentifiedViewSystem, MaybeVisualizableEntities, QueryContext, TypedComponentFallbackProvider,
+    ViewContext, ViewContextCollection, ViewQuery, ViewSystemExecutionError, VisualizableEntities,
+    VisualizableFilterContext, VisualizerQueryInfo, VisualizerSystem, auto_color_for_entity_path,
+};
+
+use crate::{contexts::SpatialSceneEntityContext, proc_mesh, view_kind::SpatialViewKind};
+
+use super::{
+    SpatialViewVisualizerData, filter_visualizable_3d_entities,
+    utilities::{ProcMeshBatch, ProcMeshDrawableBuilder},
+};
+
+// ---
+
+pub struct Cylinders3DVisualizer(SpatialViewVisualizerData);
+
+impl Default for Cylinders3DVisualizer {
+    fn default() -> Self {
+        Self(SpatialViewVisualizerData::new(Some(
+            SpatialViewKind::ThreeD,
+        )))
+    }
+}
+
+// NOTE: Do not put profile scopes in these methods. They are called for all entities and all
+// timestamps within a time range -- it's _a lot_.
+impl Cylinders3DVisualizer {
+    fn process_data<'a>(
+        builder: &mut ProcMeshDrawableBuilder<'_, Fallback>,
+        query_context: &QueryContext<'_>,
+        ent_context: &SpatialSceneEntityContext<'_>,
+        batches: impl Iterator<Item = Cylinders3DComponentData<'a>>,
+    ) -> Result<(), ViewSystemExecutionError> {
+        for batch in batches {
+            // Number of instances is determined by whichever is *longer* of `lengths` and `radii`.
+            // The other component is clamped (last value repeated) to match.
+            let num_instances = batch.radii.len().max(batch.lengths.len());
+            let lengths_iter = clamped_or_nothing(batch.lengths, num_instances);
+            let radii_iter = clamped_or_nothing(batch.radii, num_instances);
+
+            let half_sizes: Vec<HalfSize3D> = lengths_iter
+                .zip(radii_iter)
+                .map(|(&Length(length), &Radius(radius))| {
+                    let radius = clean_length(radius.0);
+                    HalfSize3D::new(radius, radius, length.0 / 2.0)
```

Comment:
worth a note why this has to be /2.0 (but following the code in this PR this makes sense!)